src/Frontend/components/portal/login/Form.js

```diff
@@ -0,0 +1,170 @@
+/*
+ * This file is part of the Sententiaregum project.
+ *
+ * (c) Maximilian Bosch <maximilian.bosch.27@gmail.com>
+ * (c) Ben Bieler <benjaminbieler2014@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+'use strict';
+
+import React, { Component } from 'react';
+import LoadableButtonBar from '../../form/LoadableButtonBar';
+import { authenticate } from '../../../actions/PortalActions';
+import AuthenticationStore from '../../../store/AuthenticationStore';
+import DismissableAlertBox from '../../app/markup/DismissableAlertBox';
+import FormHelper from '../../../util/react/FormHelper';
+import { runAction, connector } from 'sententiaregum-flux-container';
+import deepAssign from 'deep-assign';
+import FormField from '../../form/FormField';
+import counterpart from 'counterpart';
+import getStateValue from '../../../store/provider/getStateValue';
+import CurrentLocaleStore from '../../../store/CurrentLocaleStore';
+
+/**
+ * Form component for the login form.
+ *
+ * @author Maximilian Bosch <maximilian.bosch.27@gmail.com>
+ */
+export default class Form extends Component {
+  /**
+   * Constructor.
+   *
+   * @param {Object} props Component properties.
+   *
+   * @returns {void}
+   */
+  constructor(props) {
+    super(props);
+    this.handler     = this._change.bind(this);
+    this.i18nHandler = () => this.forceUpdate();
+
+    const state = AuthenticationStore.getState().message;
+    this.helper = new FormHelper(
+      { username: '' },
+      { password: '' },
+      {},
+      nextState => this.setState(deepAssign({ data: this.state.data }, nextState)),
+      'pages.portal.login.form',
+      false
+    );
+
+    this.state = this.helper.getInitialState(state ? state : {});
+  }
+
+  /**
+   * Mounts the change listeners on the store.
+   *
+   * @returns {void}
+   */
+  componentDidMount() {
+    counterpart.onLocaleChange(this.i18nHandler);
+    connector(AuthenticationStore).useWith(this.handler);
+  }
+
+  /**
+   * Removes the change listener.
+   *
+   * @returns {void}
+   */
+  componentWillUnmount() {
+    counterpart.offLocaleChange(this.i18nHandler);
+    connector(AuthenticationStore).unsubscribe(this.handler);
+  }
+
+  /**
+   * Renders the dom markup.
+   *
+   * @returns {React.Element} The markup.
+   */
+  render() {
+    let errorBox;
+    if (!this.state.success && this.helper.isSubmitted()) {
+      errorBox = (
+        <DismissableAlertBox bsStyle="danger">
+          <p>{this._getError()}</p>
```

Comment:
we could move this sort of error box into a custom file and then we don't have this error rendering logic here and the counterpart hook is unnecessary, too.